src/cf.data.pre

```diff
@@ -7112,11 +7112,22 @@ LOC: Config.visibleHostname
 DEFAULT: none
 DEFAULT_DOC: Automatically detect the system host name
 DOC_START
-	If you want to present a special hostname in error messages, etc,
-	define this.  Otherwise, the return value of gethostname()
-	will be used. If you have multiple caches in a cluster and
-	get errors about IP-forwarding you must set them to have individual
-	names with this setting.
+	The public domain name (FQDN) of this Proxy.
```

Comment:
I do not think this name has to be "public" or even fully qualified. In many deployments, it is and will be neither.

I recommend describing what this name is used for without unnecessarily (or misleadingly) restricting its scope.

Reply:
The name configured here does get used in messages going out into the public Internet, so admin wishing to keep their internal naming scheme private need to be made aware of that and set this to something that is suitable for public display.

We do see a lot of configs using internal naming and complaining that Squid publishes this detail - if they set it appropriately to begin with things like Via would not be such a big privacy/security "problem".

Reply:
> The name configured here does get used in messages going out into the public Internet

If you want to warn the admins about leaking naming information to the "public internet", please do so, but "public domain name of this Proxy" conveys a completely different message to many (probably most) readers.

Besides, not all proxies communicate with the public internet.

I continue to recommend describing what this name is used for and letting the admin decide what name they want to use for those purposes. Based on the other updated text, calling this name "host name" or "domain name" is sufficient for this context.

Reply:
> > The name configured here does get used in messages going out into the public Internet
> 
> If you want to warn the admins about leaking naming information to the "public internet", please do so, but "public domain name of this Proxy" conveys a completely different message to many (probably most) readers.
> 
> Besides, not all proxies communicate with the public internet.

Irrelevant. Most do and this directive is implemented for use on Internet.

> 
> I continue to recommend describing what this name is used for and letting the admin decide what name they want to use for those purposes.

This directive is used as an FQDN in URLs (as documented already). Other options use this name as a default value. Admin can set those other options if they have specific needs in those other use-cases.


> Based on the other updated text, calling this name "host name" or "domain name" is sufficient for this context.

That you use "host name" indicates a failure of clarity and why I am trying to be explicit about its FQDN nature. Setting it to a "host name" is a malformed config which causes invalid URI (possibly dangerous) to be transmitted. It can be a "hostname" (note lack of space in the term) - which is a valid FQDN in the networks without public DNS or traffic.

Reply:
> Alex: not all proxies communicate with the public internet.

> Amos: Irrelevant. Most do and this directive is implemented for use on Internet.

The fact that some proxies (that correctly use this directive) do not communicate with the public internet is certainly relevant to the directive description. When a directive can be successfully/correctly used for A and B, we should not describe it as "used for A". I cannot accept the "implemented for A [and not B]" argument (if that is what you are implying) either -- in this context, it is not very important what specific use cases this directive was implemented for (and I very much doubt it was implemented to exclude private network use cases).

> Alex: I continue to recommend describing what this name is used for and letting the admin decide what name they want to use for those purposes.

> Amos: This directive is used as an FQDN in URLs

Based on your statements quoted further below, I assume that here you count dotless hostnames in private networks as FQDNs. If so, you may be technically correct (you did not substantiate your claims, and I did not check myself), but that is not how most readers would interpret the above phrase. We should not use misleading (to most readers) phrasing in a directive documentation even if it that phrasing is technically accurate.

> Other options use this name as a default value. Admin can set those other options if they have specific needs in those other use-cases.

Yes, of course. IMO, it would be good to reference those other options in this directive documentation. I found two directives that explicitly refer to visible_hostname (unique_hostname and httpd_accel_surrogate_id) but perhaps there are more. If it is too difficult to provide a comprehensive list, we can just warn the reader that there are some undocumented dependencies. This is not what _this_ change request is about though.

> That you use "host name" indicates a failure of clarity and why I am trying to be explicit about its FQDN nature ... It can be a "hostname" (note lack of space in the term) - which is a valid FQDN in the networks without public DNS

Using "hostname" (without the space) is fine with me! And if you really want to add some text explaining why this hostname should be an FQDN, please feel free do so. Just assume that most readers would interpret FQDN as "a domain name ending with a TLD" or, worse, "a domain name with dots in it". IMHO, squid.conf.documented is not the place to educate readers about other FQDN meanings. We should not expect them to know those other meanings. And we should not assume they actually expect those other meanings in basic squid.conf.documented phrases even if they know about those other meanings.

To summarize, let's drop "public" and "FQDN" from the brief directive description. Let's document how this hostname is used by Squid. Adding followup paragraphs highlighting the dangers of not using an FQDNs for this parameter, especially on public networks, is perfectly fine, of course.

Reply:
Err... This is actually a thread in an old review made obsolete but the later [review](https://github.com/squid-cache/squid/pull/302#pullrequestreview-173094053) and branch changes. I do not know why you resumed this argument. FWIW, the current `The hostname of this proxy` brief description is fine with me and has been so since Nov 8, 2018!

Reply:
Was working through the open discussions. Resolving this one now since current text is fine.